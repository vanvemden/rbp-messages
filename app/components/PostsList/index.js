import React from 'react';
import PropTypes from 'prop-types';

import List from 'components/List';
import ListItem from 'components/ListItem';
import LoadingIndicator from 'components/LoadingIndicator';
import PostListItem from 'containers/PostListItem';

function PostsList({ loading, storing, error, posts }) {
  if (loading || storing) {
    return <List component={LoadingIndicator} />;
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <ListItem item="Something went wrong, please try again!" />
    );
    return <List component={ErrorComponent} />;
  }

  if (posts !== false) {
    return <List items={posts} component={PostListItem} />;
  }

  return null;
}

PostsList.propTypes = {
  loading: PropTypes.bool,
  storing: PropTypes.bool,
  error: PropTypes.any,
  posts: PropTypes.any,
};

export default PostsList;
