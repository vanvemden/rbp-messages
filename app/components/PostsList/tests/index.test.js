import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from 'react-testing-library';

import PostsList from '../index';

describe('<PostsList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<PostsList loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <PostsList loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });
});
