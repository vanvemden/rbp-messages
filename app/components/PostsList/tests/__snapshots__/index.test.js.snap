// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsList /> should render the loading indicator when its loading 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <div
      class="Wrapper-sc-12uw37d-0 bTWIVF"
    >
      <div
        class="Circle__CirclePrimitive-ww56dy-0 enQIMS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jCxmwS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 dyFXB"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ikwUeP"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hwlQRz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jjXxyb"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jVExkJ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 haCVIz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ghhiLZ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 bCbqdt"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hZPQMA"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 fsGOmP"
      />
    </div>
  </ul>
</div>
`;
